aries_cloudagent/core/dispatcher.py

```diff
@@ -377,7 +377,28 @@ async def create_outbound(
 
         return await super().create_outbound(message, **kwargs)
 
-    async def send_outbound(self, message: OutboundMessage) -> OutboundSendStatus:
+    async def _get_msg_type_from_enc_payload(
+        self, profile: Profile, parsed_msg: dict
+    ) -> Optional[Tuple[str, str]]:
+        """Get message type and id tuple from enc_payload."""
+        try:
+            if not isinstance(parsed_msg, dict):
+                return None
+            message_type = parsed_msg.get("@type")
+            if not message_type:
+                return None
+            registry: ProtocolRegistry = profile.inject(ProtocolRegistry)
+            message_cls = registry.resolve_message_class(message_type)
+            if not message_cls:
+                return None
+            instance = message_cls.deserialize(parsed_msg)
+            return instance._message_type, instance._id
+        except (ProtocolMinorVersionNotSupported, BaseModelError, AttributeError):
+            return None
```

Comment:
It would have been necessary in `aries_cloudagent/messaging/responder.py` `send()` and `send_reply()`, when message can be instance of `str` or `bytes` but the following cleanup can done.

How about updating `aries_cloudagent/messaging/responder.py` `send()` and `send_reply()` as:
```
if isinstance(message, BaseMessage):
    msg_type = message._message_type
    msg_id = message._id
else:
    msg_dict = json.loads(message)
    msg_type = msg_dict.get("@type")
    msg_id = msg_dict.get("@id")
return await self.send_outbound(
        message=outbound,
        message_type=msg_type,
        message_id=msg_id,
    )
```

And removing `_get_msg_type_from_enc_payload` and the following code from `./aries_cloudagent/core/dispatcher.py` `send_outbound()`:
```
if not msg_type and not msg_id and (message.enc_payload or message.payload):
        msg_dict = json.loads(message.enc_payload or message.payload)
        msg_type_id_tuple = await self._get_msg_type_from_enc_payload(
            context.profile, msg_dict
        )
        if msg_type_id_tuple:
            msg_type, msg_id = msg_type_id_tuple
```

Reply:
I think the changes you propose would be a good improvement, keeping the changes closer to the scenario that makes them necessary.